src/utility/SHT3x.cpp

```diff
@@ -1,542 +1,149 @@
-    SetAddress((uint8_t)Address);
-    _ValueIfError = Value;
-    SetMode(Mode);
-    _SensorType = SensorType;
-    _HardResetPin = HardResetPin;
-}
-
-void SHT3x::Begin()
-{
-    Wire.begin(21, 22);
-    _OperationEnabled = true;
-}
+#include <Wire.h>
 
-void SHT3x::UpdateData()
+namespace
 {
-    _Error = noError;
-    if ((_LastUpdateMillisec == 0) || ((millis() - _LastUpdateMillisec) >= _UpdateIntervalMillisec))
-    {
-        SendCommand(_MeasMSB, _MeasLSB);
-        if (_Error == noError)
-        {
-            Wire.requestFrom(_Address, (uint8_t)6);
-            uint32_t WaitingBeginTime = millis();
-            while ((Wire.available() < 6) && ((millis() - WaitingBeginTime) < _TimeoutMillisec))
-            {
-                //Do nothing, just wait
-            }
-            if ((millis() - WaitingBeginTime) < _TimeoutMillisec)
-            {
-                _LastUpdateMillisec = WaitingBeginTime;
-                uint8_t data[6];
-                for (uint8_t i = 0; i < 6; i++)
-                {
-                    data[i] = Wire.read();
-                }
-
-                if ((CRC8(data[0], data[1], data[2])) && (CRC8(data[3], data[4], data[5])))
-                {
-                    uint16_t TemperatureRaw = (data[0] << 8) + (data[1] << 0);
-                    uint16_t RelHumidityRaw = (data[3] << 8) + (data[4] << 0);
-                    _TemperatureCeil = ((float)TemperatureRaw) * 0.00267033 - 45.;
-                    _TemperatureCeil = _TemperatureCeil * _TemperatureCalibration.Factor +
-                                       _TemperatureCalibration.Shift;
-                    _RelHumidity = ((float)RelHumidityRaw) * 0.0015259;
-                    _RelHumidity = _RelHumidity * _RelHumidityCalibration.Factor +
-                                   _RelHumidityCalibration.Shift;
-
-                    _Error = noError;
-                }
-                else
-                {
-                    _Error = DataCorrupted;
-                }
-            }
-            else //Timeout
-            {
-                _Error = Timeout;
-            }
-        }
-        else //Error after message send
-        {
-            // Nothing to do, measurement commands will return NULL because of Error != noError
-        }
-    }
-    else //LastUpdate was too recently
-    {
-        //Nothing to do, wait for next call
-    }
-}
+    const uint16_t I2C_ADDRESS = 0x44;
 
-float SHT3x::GetTemperature(TemperatureScale Degree)
-{
-    //default scale is Celsius
-    //At first, calculate in Celsius, than, if need, recalculate to Farenheit or Kelvin and than adjust according to calibration;
-    float Temperature = _TemperatureCeil;
-    if (Degree == Kel)
-    {
-        Temperature += 273.15;
-    }
-    else if (Degree == Far)
-    {
-        Temperature = Temperature * 1.8 + 32.;
-    }
+    // Repatability: High, Clock stretching: Enabled
+    const uint8_t MEASUREMENT_MSB = 0x2c;
+    const uint8_t MEASUREMENT_LSB = 0x06;
```

Comment:
I'm not sure this is the best measurement command. (I'm not sure what repeatability and clock stretching means.)
I used the first one in the datasheet.